packages/core/lib/callback-runner.js

```diff
@@ -0,0 +1,46 @@
+const some = require('./async-every')
+
+module.exports = (callbacks, event, onCallbackError, cb) => {
+  // This function is how we support different kinds of callback:
+  //  - synchronous - return value
+  //  - node-style async with callback - cb(err, value)
+  //  - promise/thenable - resolve(value)
+  // It normalises each of these into the lowest common denominator – a node-style callback
+  const runMaybeAsyncCallback = (fn, cb) => {
+    if (typeof fn !== 'function') return cb(null, true)
+    try {
+      // if function appears sync…
+      if (fn.length !== 2) {
+        const ret = fn(event)
+        // check if it returned a "thenable" (promise)
+        if (ret && typeof ret.then === 'function') {
+          return ret.then(
+            // resolve
+            val => setTimeout(() => cb(null, val), 0),
+            // reject
+            err => {
+              setTimeout(() => {
+                onCallbackError(err)
```

Comment:
`onCallbackError` is not covered by unit tests

Reply:
Added a test for this.